tests/container_test.go

```diff
@@ -804,3 +804,44 @@ func stackWithBool(t *testing.T, stack *vm.Stack, v bool) {
 	require.True(t, ok)
 	require.Equal(t, v, res)
 }
+
+func TestPutMeta(t *testing.T) {
+	c, cBal, _ := newContainerInvoker(t, false)
+
+	t.Run("meta disabled", func(t *testing.T) {
+		acc := c.NewAccount(t)
+		cnt := dummyContainer(acc)
+		balanceMint(t, cBal, acc, containerFee*1, []byte{})
+
+		metaInfo, err := stackitem.Serialize(stackitem.NewMapWithValue(
+			[]stackitem.MapElement{{Key: stackitem.Make("cid"), Value: stackitem.Make(cnt.id[:])}}))
+		require.NoError(t, err)
+
+		c.Invoke(t, stackitem.Null{}, "put", cnt.value, cnt.sig, cnt.pub, cnt.token, false)
+		c.InvokeFail(t, "container does not support meta-on-chain", "putObject", metaInfo, nil)
+
+		expected := stackitem.NewStruct([]stackitem.Item{
+			stackitem.NewByteArray(cnt.value),
+			stackitem.NewByteArray(cnt.sig),
+			stackitem.NewByteArray(cnt.pub),
+			stackitem.NewByteArray(cnt.token),
+		})
+		c.Invoke(t, expected, "get", cnt.id[:])
+	})
+
+	t.Run("meta enabled", func(t *testing.T) {
+		acc := c.NewAccount(t)
+		cnt := dummyContainer(acc)
+		balanceMint(t, cBal, acc, containerFee*1, []byte{})
+
+		metaInfo, err := stackitem.Serialize(stackitem.NewMapWithValue(
+			[]stackitem.MapElement{
+				{Key: stackitem.Make("cid"), Value: stackitem.Make(cnt.id[:])},
+				{Key: stackitem.Make("oid"), Value: stackitem.Make(cnt.id[:])},
+			}))
+		require.NoError(t, err)
+
+		c.Invoke(t, stackitem.Null{}, "put", cnt.value, cnt.sig, cnt.pub, cnt.token, true)
+		c.Invoke(t, stackitem.Null{}, "putObject", metaInfo, nil)
```

Comment:
why is it OK w/o signatures?

Reply:
for test purposes, container was added without any nodes (again, IMO signature verification should not be tested in this PR), so no nodes can add its signature but also no nodes are quired to do it, every iterator over node's put keys does nothing